src/generic/Matrix.jl

```diff
@@ -232,6 +232,11 @@ zero(x::MatrixElem, R::Ring, r::Int, c::Int) = zero!(similar(x, R, r, c))
 zero(x::MatrixElem, R::Ring=base_ring(x)) = zero(x, R, nrows(x), ncols(x))
 zero(x::MatrixElem, r::Int, c::Int) = zero(x, base_ring(x), r, c)
 
+@doc Markdown.doc"""
```

Comment:
I don't personally object to this, but technically zero is documented elsewhere in the system and there is currently a push to deduplicate the documentation, rather than to introduce many new docstrings for the same thing.

Reply:
Are you referring to [this](https://nemocas.github.io/AbstractAlgebra.jl/latest/module/#Base.zero-Union{Tuple{AbstractAlgebra.FPModule{T}},%20Tuple{T}}%20where%20T%3C:Union{RingElem,%20AbstractFloat,%20Integer,%20Rational})?

In relation to Nemo (it links to https://nemocas.github.io/AbstractAlgebra.jl/latest/matrix/), could it be valuable to have the `zero` function under `matrix.md` as well?

Reply:
I wasn't referring to that. The zero function is not documented there. It is documented as an optional part of the Matrix interface. It's optional for a reason.

If you look on the matrix.md page, you will see it is already documented:

"As well as implementing the entire Matrix interface, including the optional functionality, there are many additional generic algorithms implemented for matrix spaces. We describe this functionality below."

It is current wisdom that we should not duplicate documentation, so although I wouldn't object to putting this in, I suspect it will simply be removed by people currently on a deduplication push. Perhaps wait until the pendulum swings the other way.

Reply:
As one of the people @wbhart refers to regarding the "deduplication push": what "we" object to is having the help function showing dozens or hundreds of near identical doc strings for things like `+`. I'd rather have only one in a central place (or three, or five, if there are different signatures to document; but the point: document each variant only once).

As such, I think I'd probably prefer if all functions of a given implicit "interface" were documented in one place -- so I'd not have suggested to remove the `zero_matrix` docstring above, even if there is no generic implementation...